app/src/main/java/com/github/multimatum_team/multimatum/adaptater/DeadlineAdapter.kt

```diff
@@ -108,15 +115,34 @@ class DeadlineAdapter(private val context: Context) : BaseAdapter() {
         subtitleTextView.setTypeface(null, Typeface.ITALIC)
 
         // Show how much time left or if it is due or done.
+        updateDetails(detailTextView, deadline)
+
+        // Set the checkbox
+        doneCheckbox.isChecked = (deadline.state == DeadlineState.DONE)
+        doneCheckbox.setOnCheckedChangeListener { _, isChecked ->
+            val newDeadline = Deadline(
+                deadline.title,
+                if (isChecked) DeadlineState.DONE else DeadlineState.TODO,
+                deadline.dateTime
+            )
```

Comment:
You can use `deadline.copy(state = if (isChecked) DeadlineState.DONE else DeadlineState.TODO)` instead (kotlin data classes automatically generates the `copy` method)